akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -429,22 +429,19 @@ final class Split[T](val decision: Split.SplitDecision, val p: T ⇒ Boolean, va
 
     setHandler(out, new OutHandler {
       override def onPull(): Unit = {
-        if (substreamSource eq null) pull(in)
-        else if (!substreamWaitingToBePushed) {
-          push(out, Source.fromGraph(substreamSource.source))
-          scheduleOnce(SubscriptionTimer, timeout)
-          substreamWaitingToBePushed = true
-        }
+        if (substreamSource eq null) {
+          //can be already pulled from substream in case split after
+          if (!hasBeenPulled(in)) pull(in)
+        } else if (substreamWaitingToBePushed) pushSubstreamSource()
```

Comment:
@drewhk  the name of the variable should be "substreamWasPushed" in legacy code because of following:
```
if (!substreamWaitingToBePushed) {
 push(out, Source.fromGraph(substreamSource.source))
...
 substreamWaitingToBePushed = true
}
```
There are three stages - `substreamSource` is null because data not yet come from upstream.
When data came it can be pushed downstream or not yet pushed so `substreamWaitingToBePushed` should be true in this state but it's false. And the third stage when substream is pushed.
My assumption is that `substreamWaitingToBePushed` was copied from `groupBy` stage with some misconception.
Let me know if I missed something.